setup.py

```diff
@@ -135,27 +137,77 @@ def get_ext_modules():
                                             "numba/_pymodule.h"],
                                    **np_compile_args)
 
+
+    ext_npyufunc_workqueue_impls = []
+
+    # Search for Intel TBB, first check env var TBBROOT then conda locations
     tbb_root = os.getenv('TBBROOT')
+    if not tbb_root:
+        path2check = [os.path.split(os.path.split(sys.executable)[0])[0]]
+        path2check += [os.getenv(n, '') for n in ['CONDA_PREFIX', 'PREFIX']]
+        if sys.platform.startswith('win'):
+            path2check += [os.path.join(p, 'Library') for p in path2check]
+        for p in path2check:
+            if p and os.path.isfile(os.path.join(p, 'include', 'tbb', 'tbb.h')):
+                tbb_root = p  # the latest is used
+
+
+    if sys.platform.startswith('win'):
+        cpp11flags = []
+        ompcompileflags = ['-openmp']
+        omplinkflags = []
+    elif sys.platform.startswith('darwin'):
+        cpp11flags = ['-std=c++11']
+        ompcompileflags = ['-fopenmp']
+        omplinkflags = ['-fopenmp=libomp']
+    else:
+        cpp11flags = ['-std=c++11']
+        ompcompileflags = ['-fopenmp']
+        if platform.machine() == 'ppc64le':
+            omplinkflags = ['-fopenmp']
+        else:
+            omplinkflags = ['-fopenmp']
 
     if tbb_root:
-        print("Using TBBROOT=", tbb_root)
-        ext_npyufunc_workqueue = Extension(
-            name='numba.npyufunc.workqueue',
-            sources=['numba/npyufunc/tbbpool.cpp', 'numba/npyufunc/gufunc_scheduler.cpp'],
-            depends=['numba/npyufunc/workqueue.h'],
-            include_dirs=[os.path.join(tbb_root, 'include')],
-            extra_compile_args=[] if sys.platform.startswith('win') else ['-std=c++11'],
-            libraries   =['tbb'],
-            library_dirs=[os.path.join(tbb_root, 'lib', 'intel64', 'gcc4.4'),  # for Linux
-                          os.path.join(tbb_root, 'lib'),                       # for MacOS
-                          os.path.join(tbb_root, 'lib', 'intel64', 'vc_mt'),   # for Windows
-                         ],
-            )
-    else:
-        ext_npyufunc_workqueue = Extension(
-            name='numba.npyufunc.workqueue',
-            sources=['numba/npyufunc/workqueue.c', 'numba/npyufunc/gufunc_scheduler.cpp'],
-            depends=['numba/npyufunc/workqueue.h'])
+        # Python 2 on windows has a VS dep which cannot handle sufficiently
+        # advanced C++ syntax to compile tbbpool.cpp.
+        tbb_unsupported = sys.platform.startswith('win') and \
+                          sys.version_info < (3, 0)
```

Comment:
could this be specified in recipe instead? E.g. if no tbb is present, no `TBBROOT` is specified, then it will not be compiled anyway, no need for additional check. Otherwise, it breaks Intel's build (unless patched) since we use icc 13 which can compile it for py2win

Reply:
Yes. I think that is possible and essentially the same effort, good point about ICC.